### PR TITLE
[grafana] Improve ingress API logic

### PR DESCRIPTION
**Pull request by stevehipwell**
This PR improves the logic in #505.

### CODE REVIEW

File: charts/grafana/templates/_helpers.tpl [hunk 1]
```diff
@@ -139,7 +132,7 @@ Return the appropriate apiVersion for rbac.
 Return the appropriate apiVersion for ingress.
 */}}
 {{- define "grafana.ingress.apiVersion" -}}
-  {{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19.x" (include "grafana.kubeVersion" .)) -}}
+  {{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19-0" .Capabilities.KubeVersion.Version) -}}
```
Comment (torstenwalter) [inline charts/grafana/templates/_helpers.tpl:135]: Where do I find an explanation for the `-0`?

Comment (stevehipwell) [reply charts/grafana/templates/_helpers.tpl:135]: It's documented in the [Helm docs](https://helm.sh/docs/chart_template_guide/function_list/#semvercompare), but the short explanation is that `-0` is the lowest possible pre-release version. After I made the original changes in #505 I spent a bit of time looking around and this pattern seems to be idiomatic and once I figured out what it was doing I saw that it was the correct solution for where we want to constrain on a K8s release line.

**Review by zanhsieh — APPROVED** [review]

**Comment by naimo84** [comment]
Hey ho, 

I have similar problems like here: https://github.com/argoproj/argo-helm/issues/753

If you use argocd to deploy your Helm Chart the .Capabilities.KubeVersion.Version is v1.20.0 but the actual k8s cluster version is v1.18.0

So the ingress apiVersion is wrong :(

Is it possible to introduce a "kubeVersionOverride"-like property?

```yaml
argo-workflows:
  kubeVersionOverride: "1.18-3"
```

Currently I can't use the grafana helm chart.

Greets,
Benjamin

**Comment by stevehipwell** [comment]
@naimo84 that sounds like an Argo defect to me. Do they not have the equivalent of the Helm flag to set the cluster version?

**Comment by naimo84** [comment]
@stevehipwell unfortunatelly not 😐 as argumented here: 
https://github.com/argoproj/argo-cd/issues/5202#issuecomment-756406140

only a "workaround" is possible

**Comment by stevehipwell** [comment]
@naimo84 I'm referring to the `--kube-version' flag which doesn't need any cluster credentials. When I updated the ingress templates I made sure they'd work like this.

**Comment by naimo84** [comment]
@stevehipwell many many thanks for pointing me the right way 👍 it's Part of the next version https://github.com/argoproj/argo-cd/pull/6359
and thanks for your great work regarding the templates 😀

**Pull request merged**